### PR TITLE
Proxy

### PR DESCRIPTION
**Pull request by joaquin386**
#### Pull Request (PR) description
Adding Proxy to be able to download packages behind a proxy (normal company scenario)

#### This Pull Request (PR) fixes the following issues
Puppet/Servers behind a Proxy

### CODE REVIEW

File: manifests/beanstalkd_exporter.pp [hunk 1]
```diff
@@ -105,6 +109,7 @@
   Boolean $manage_user                = true,
   String $os                          = $prometheus::os,
   String $extra_options               = '',
+  Optional[String] $proxy             = undef,
```
Comment (bastelfreak) [inline manifests/beanstalkd_exporter.pp]: can you please use `Optional[String[1]]`?

**Comment by bastelfreak** [comment]
@joaquin386 thanks for the PR. Can you please rebase?

File: manifests/init.pp [hunk 2]
```diff
@@ -136,6 +143,7 @@
   Stdlib::Absolutepath $shared_dir,
   String $version,
   String $install_method,
+  Optional[String] $proxy,
```
[the remark this answers is not in the document]
Comment (joaquin386) [reply manifests/init.pp]: it appear twice, maybe I miss the top one but on v2 is removed.

File: manifests/init.pp [hunk 3]
```diff
@@ -154,6 +162,7 @@
   Array $alertmanagers_config,
   String $storage_retention,
   Stdlib::Absolutepath $env_file_path,
+  Optional[String] $proxy        = undef,
```
[the remark this answers is not in the document]
Comment (joaquin386) [reply manifests/init.pp]: I must have miss the top one. Now erase the one that doesn't have the undef on v2.

File: manifests/init.pp [hunk 4]
```diff
@@ -34,6 +34,13 @@
 #  [*install_method*]
 #  Installation method: url or package (only url is supported currently)
 #
+#  [*proxy*]
+#  Will set the http_proxy and https_proxy env variables in 
+#  (example: proxy => 'http://proxy.company.com:8080) 
+#
+#  [*no_proxy*]
```
[the remark this answers is not in the document]
Comment (joaquin386) [reply manifests/init.pp:41]: v2 implemented but the no_proxy left in there for future development.

File: manifests/beanstalkd_exporter.pp (hunk 1 above)
Comment (joaquin386) [reply manifests/beanstalkd_exporter.pp]: Put the [1], dunno which one is better.

**Comment by bastelfreak** [comment]
@joaquin386 can you please rebase and check the used email address in the commits? They aren't associated with your github account.

**Comment by joaquin386** [comment]
i can not do it sooner than 3 weeks. Right now between jobs. i can close the pr and when ready i can re open it.

**Comment by bastelfreak** [comment]
sounds good. Just rebase and reopen it whenever you've time for it.

**Pull request closed without merging**